languages/ruby/dyp/il_ruby_build.ml

```diff
@@ -35,7 +35,7 @@ let acc_append acc1 acc2 =
     q = DQueue.append acc1.q acc2.q;
     seen = StrSet.union acc1.seen acc2.seen;
     super_args =
-      (assert (acc1.super_args == acc2.super_args);
+      (assert (phys_equal acc1.super_args acc2.super_args);
        acc1.super_args);
   }
```

Comment:
semgrep bug? Should have been caught by https://semgrep.dev/playground/r/ocaml.lang.correctness.physical-vs-structural.physical-equal

Reply:
excluded from the scan by semgrep.jsonnet